### ISSUE TITLE
help or docs on upgrading to Monterey

### ISSUE THREAD

**Issue by joaomcferreira** [myspaghetti/macos-virtualbox#583]
Hello,

Thank you for this great project. I have installed it several times with success and it is fine. But I have also tried many times to upgrade to Monterey (following several indications like number of processors and using an alternative storage for Monterey) but it always failed.

It would be perfect for me if someone could help me or some document be written on how to achieve this. I tried several times and it always fails. I would love to use it with Monterey due to a personal project.

Thank you
Joao

**Comment by myspaghetti** [comment]
The best advice is to run it on QEMU/KVM if you can. That gives you fine-grained control over the CPUID and better compatibility in general. VirtualBox has fine-grained control over CPUID too, but it has to be entered in hex which is a pain, while QEMU/KVM uses [mnemonics](https://qemu-project.gitlab.io/qemu/system/qemu-cpu-models.html) which [simplifies turning specific features on and off.](https://github.com/kholia/OSX-KVM/blob/master/OpenCore-Boot-macOS.sh)

Thanks to WSLg, [QEMU/KVM is even available on Windows 10 and 11.](https://boxofcables.dev/kvm-optimized-custom-kernel-wsl2-2022/) If you want the absolute best performance, you might be better off running Linux on bare metal (maybe [Proxmox](https://www.nicksherlock.com/2021/10/installing-macos-12-monterey-on-proxmox-7/)) and running Windows and macOS on QEMU.

As for VirtualBox, I've found that downloading Monterey from Software Update or the app store and installing it on an empty volume on its own virtual disk works, as long as you keep the script's default 2 CPU cores. Others have reported that Monterey crashes every few hours or when running a specific app.

TL;DR if you already have a working macOS disk image, put it on QEMU/KVM, and upgrade to Monterey. You could run it on VirtualBox but not very well.

**Comment by myspaghetti** [comment]
The best advice is to run it on QEMU/KVM if you can. That gives you fine-grained control over the CPUID and better compatibility in general. VirtualBox has fine-grained control over CPUID too, but it has to be entered in hex which is a pain, while QEMU/KVM uses [mnemonics](https://qemu-project.gitlab.io/qemu/system/qemu-cpu-models.html) which [simplifies turning specific features on and off.](https://github.com/kholia/OSX-KVM/blob/master/OpenCore-Boot-macOS.sh)

Thanks to WSLg, [QEMU/KVM is even available on Windows 10 and 11](https://boxofcables.dev/accelerated-kvm-guests-on-wsl-2/) (you can skip all the compulation stages if `cat /sys/module/kvm_intel/parameters/nested` outputs `Y`). If you want the absolute best performance, you might be better off running Linux on bare metal (maybe [Proxmox](https://www.nicksherlock.com/2021/10/installing-macos-12-monterey-on-proxmox-7/)) and running Windows and macOS on QEMU.

As for VirtualBox, I've found that downloading Monterey from Software Update or the app store and installing it on an empty volume on its own virtual disk works, as long as you keep the script's default 2 CPU cores. Others have reported that Monterey crashes every few hours or when running a specific app.

TL;DR if you already have a working macOS disk image, put it on QEMU/KVM, and upgrade to Monterey. You could run it on VirtualBox but not very well.

**Comment by UInt2048** [comment]
> TL;DR if you already have a working macOS disk image, put it on QEMU/KVM

Do you have any resources to make that step as straightforward as you made it seem?

**Comment by UInt2048** [comment]
> TL;DR if you already have a working macOS disk image, put it on QEMU/KVM

Do you have any resources to make that step as straightforward as you made it seem? According to the README, QEMU and KVM require additional configuration that is beyond the scope of the script.

**Comment by myspaghetti** [comment]
It's not straightforward but it's the best solution. Install OpenCore (on a separate image or the same image), configure QEMU, and boot. Basically what the ProxMox guide says except instead of booting an installer, you boot an existing image.

**Comment by UInt2048** [comment]
@myspaghetti Thank you for the information. I'm using a macOS host but it seems like these guides can only be followed on Linux. Even if it means starting from scratch, is there a way to do this on a real Mac (e.g. for use in UTM) without having to use a bootable Linux USB?

**Comment by Masamune3210** [comment]
Apple computers shouldn't need this script as from my understanding their versions of VirtualBox/VMware/inseremulationsoftwarehere could already do it natively

**Comment by ArsenicBismuth** [comment]
> As for VirtualBox, I've found that downloading Monterey from Software Update or the app store and installing it on an empty volume on its own virtual disk works, as long as you keep the script's default 2 CPU cores. Others have reported that Monterey crashes every few hours or when running a specific app.

@myspaghetti Finally successfully upgraded to Monterey by setting 1 CPU cores and installing on another volume. But seems to be unsuccessful in installing Xcode (while other functionalities work).
